hydro_cli_examples/examples/pn_counter/main.rs

```diff
@@ -0,0 +1,136 @@
+        .into_source();
+
+    let query_responses = ports
+        .remove("query_responses")
+        .unwrap()
+        .connect::<ConnectedBidi>()
+        .await
+        .into_sink();
+
+    let to_peer = ports
+        .remove("to_peer")
+        .unwrap()
+        .connect::<ConnectedDemux<ConnectedBidi>>()
+        .await
+        .into_sink();
+
+    let from_peer = ports
+        .remove("from_peer")
+        .unwrap()
+        .connect::<ConnectedBidi>()
+        .await
+        .into_source();
+
+    let f1 = async {
+        #[cfg(target_os = "linux")]
+        loop {
+            let x = procinfo::pid::stat_self().unwrap();
+            println!("memory: {} bytes", x.rss * 1024 * 4);
+            tokio::time::sleep(Duration::from_secs(1)).await;
+        }
+    };
+
+    let df = hydroflow_syntax! {
+        next_state = merge()
+            -> fold::<'static>((0, HashMap::<u64, (Vec<u32>, Vec<u32>)>::new()), |(_, mut cur_state), goi| {
+                match goi {
+                    GossipOrIncrement::Gossip(gossip) => {
+                        for (counter_id, (pos, neg)) in gossip.iter() {
+                            let cur_value = cur_state.entry(*counter_id).or_insert((
+                                vec![0; num_replicas], vec![0; num_replicas]
+                            ));
+
+                            for i in 0..num_replicas {
+                                cur_value.0[i] = std::cmp::max(cur_value.0[i], pos[i]);
+                                cur_value.1[i] = std::cmp::max(cur_value.1[i], neg[i]);
+                            }
+                        }
+                    }
+                    GossipOrIncrement::Increment(counter_id, delta) => {
+                        let cur_value = cur_state.entry(counter_id).or_insert((
+                            vec![0; num_replicas], vec![0; num_replicas]
+                        ));
+                        if delta > 0 {
+                            cur_value.0[my_id] += delta as u32;
+                        } else {
+                            cur_value.1[my_id] += (-delta) as u32;
+                        }
+                    }
+                }
+
+                (context.current_tick(), cur_state)
+            }) -> filter(|t| t.0 == context.current_tick())
+            -> map(|t| t.1) -> tee();
+
+        source_stream(from_peer)
+            -> map(|x| deserialize_from_bytes::<GossipOrIncrement>(&x.unwrap()).unwrap())
+            -> next_state;
+
+        source_stream(increment_requests)
+            -> map(|x| String::from_utf8(x.unwrap().to_vec()).unwrap())
+            -> map(|x| serde_json::from_str::<IncrementRequest>(&x).unwrap())
+            -> map(|t| GossipOrIncrement::Increment(t.tweet_id, t.likes))
+            -> next_state;
+
+        all_peers = repeat_iter(0..num_replicas)
+            -> filter(|x| *x != my_id);
+
+        all_peers -> [0] broadcaster;
+        next_state -> [1] broadcaster;
+        broadcaster = cross_join()
```

Comment:
This is a 'static cross join by default so it will store everything over time? Maybe the next state side should be 'tick? And the list of peers should be source_iter but then 'static in the join